### PR TITLE
Add event.dataset to events from cloudfoundry filebeat input

### LINKED ISSUE #17000: Add event.dataset to filebeat for cloud foundry fields
@blakerouse 
@mukeshelastic 

- Version: 7.7.0-SNAPSHOT
- Operating System: ALL 

Please add `event.dataset` to the exported fields with values from the `source.type` so that in the exported data field the following values will be available.
```
event.dataset = logmessage
event.dataset = httpstopstart
```

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds `event.dataset` to events generated by the `cloudfoundry` filebeat input. This is so the events match the same format that metricbeat automatically sets for events.

## Why is it important?

Helps with filtering and building dashboards for cloudfoundry.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~
- ~~I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## Related issues

- Closes elastic/beats#17000

### CODE REVIEW

**Comment by jsoriano** [comment]
I think that Filebeat inputs don't set `event.dataset`, this is set in modules (e.g. the `s3` input doesn't add `event.dataset`, but `aws` filesets add it). `input.type` is set in any case.

If we inclue this change we also have to think:
* What `event.dataset` would be set on modules that use the cloudfoundry input?
* Should we set `event.dataset` in all inputs?

Maybe as an alternative, we can set for this case `input.type` to the values proposed, and leave `event.dataset` for filesets as in other cases.

**Comment by blakerouse** [comment]
@jsoriano The field is trying to get the same field that metricbeat uses, instead of 2 different fields `event.dataset` for metricbeat and `input.type` in the other.

If `event.dataset` is only set by the module we could add a cloudfoundry module to filebeat that just sets up the input and adds the `event.dataset`. That seems like a little bit over engineering to just set one field, but maybe that would be a good base to maybe expand on the module later? But then again I think having a module might be less confusing to the user because at the moment its an input in filebeat and a module in metricbeat. So having 1 module in each, might provide a better user experience. Thoughts?

If the module is a no and don't want to set the `event.dataset` field in the input it, I could just go back to the idea of `cloudfoundry.type` on the event, that way it's the same from metricbeat and filebeat. But thats not really using a ECS field for that.

**Comment by jsoriano** [comment]
> @jsoriano The field is trying to get the same field that metricbeat uses, instead of 2 different fields `event.dataset` for metricbeat and `input.type` in the other.
> 

Yes, the problem I see is that filebeat uses `event.dataset` to indicate the module and fileset, and `input.type` to indicate the input, we should also be consistent with this. I am not against setting `event.dataset` by default to some value decided by the input, but then we should consider doing it for other inputs too.

> If `event.dataset` is only set by the module we could add a cloudfoundry module to filebeat that just sets up the input and adds the `event.dataset`. That seems like a little bit over engineering to just set one field, but maybe that would be a good base to maybe expand on the module later?

In the case of azure or AWS there are filesets for specific services or kinds of logs, could something like this make sense on cloudfoundry? In that case I think we could have a cloudfoundry module to collect data about cloudfoundry itself.

Also take into account that filebeat modules allow to override their inputs, so you could use for example the redis module with the S3 input, in this case `event.dataset` would be `redis.log` and the `input.type` would be `s3`.

If the cloudfoundry input can collect plain logs from services deployed there we could also use the redis module with this input, in this case `event.dataset` would also be `redis.log`, and `input.type` would be `cloudfoundry`.

> But then again I think having a module might be less confusing to the user because at the moment its an input in filebeat and a module in metricbeat. So having 1 module in each, might provide a better user experience. Thoughts?
> 

Yeah, I think that having a module for the use case of collecting data about cloudfoundry itself can make sense.

> If the module is a no and don't want to set the `event.dataset` field in the input it, I could just go back to the idea of `cloudfoundry.type` on the event, that way it's the same from metricbeat and filebeat. But thats not really using a ECS field for that.

If there are different kinds of events collected by the same input it can make sense to have a field specific for this input, like `cloudfoundry.type` in this case. (There is also an `event.kind` field in ECS, but it has a limited number of accepted values, and not sure if this is thought for this).

**Comment by blakerouse** [comment]
@jsoriano Now that I understand better that the `event.dataset` would be more of a module that does parsing on top of an input I have change to using `cloudfoundry.type` as the type information is coming directly from Cloud Foundry.

I think this actually makes the most sense from the input and will be shared with the module in metricbeat as well.

It's ready for another look.

File: x-pack/metricbeat/module/cloudfoundry/_meta/fields.yml [hunk 1]
```diff
@@ -8,6 +8,10 @@
       type: group
       description: >
       fields:
+        - name: type
+          type: keyword
+          description: >
+            The type of event from Cloud Foundry.
```
Comment (jsoriano) [inline x-pack/metricbeat/module/cloudfoundry/_meta/fields.yml]: Nit. Add some possible values as example to the description.

File: x-pack/libbeat/common/cloudfoundry/events.go [hunk 2]
```diff
@@ -492,9 +492,9 @@ func envelopMap(evt Event) common.MapStr {
 func baseMap(evt Event) common.MapStr {
 	return common.MapStr{
 		"cloudfoundry": common.MapStr{
+			"type": evt.String,
```
Comment (jsoriano) [inline x-pack/libbeat/common/cloudfoundry/events.go]: Should we call the function?
```suggestion
			"type": evt.String(),
```

File: x-pack/metricbeat/module/cloudfoundry/_meta/fields.yml (hunk 1 above)
Comment (blakerouse) [reply x-pack/metricbeat/module/cloudfoundry/_meta/fields.yml]: Added.

File: x-pack/libbeat/common/cloudfoundry/events.go (hunk 2 above)
Comment (blakerouse) [reply x-pack/libbeat/common/cloudfoundry/events.go]: Definitely should call the function!! Fixed.

**Comment by blakerouse** [comment]
@jsoriano Fixed, ready for another review.

**Pull request merged**